File: java_projects/catgame/src/com/quetzal/catgame/GameLogic.java
package com.quetzal.catgame;

import java.util.Scanner;

public class GameLogic {
    private char[][] board;
    private char player1Symbol;
    private char player2Symbol;

    public GameLogic() {
        board = new char[][] {
            {' ', ' ', ' '},
            {' ', ' ', ' '},
            {' ', ' ', ' '}
        };

        player1Symbol = ' ';
        player2Symbol = ' ';
    }

    public char[][] getBoard() {
        return board;
    }

    public void makeMove(int row, int col) {}

    public boolean isGameOver() {
        return true;
    }

    public void choosePlayerSymbol() {
        System.out.println("Jugador 1.\nElige tu ficha:");
        System.out.println("- X");
        System.out.println("- O");

        try (Scanner scanner = new Scanner(System.in)) {
            String symbol = scanner.nextLine().toUpperCase();
            if(symbol.equals("O") || symbol.equals("X")) {
                player1Symbol = symbol.charAt(0);
                player2Symbol = (player1Symbol == 'O') ? 'X' : 'O';
            } else {
                System.out.println("Error, ficha no valida.\nSe seleccionó al Jugador 1 la ficha por defecto: 'O' ");
                player1Symbol = 'O';
                player2Symbol = 'X';
            }

            System.out.println("Jugador 1: " + player1Symbol);
            System.out.println("Jugador 2: " + player2Symbol);
        }

        System.out.println("");
    }
}

File: python_projects/catgame/game_logic.py
class GameLogic:
    def __init__(self):
        self.boar = [
            [' ', ' ', ' '],
            [' ', ' ', ' '],
            [' ', ' ', ' ']
        ]

        self.player_1_symbol = ' '
        self.player_2_symbol = ' '


    def get_board(self):
        return self.boar


    def choose_player_symbol(self):
        print("Jugador 1.\nElige tu ficha:")
        print("- X")
        print("- O")

        symbol = input("").upper()
        if symbol == 'X' or symbol == 'O':
            self.player_1_symbol = symbol
            self.player_2_symbol = 'X' if self.player_1_symbol == 'O' else 'O'
        else:
            print("Error, ficha no valida.\nSe seleccionó al Jugador 1 la ficha por defecto: 'O' ")
            self.player_1_symbol = 'O'
            self.player_2_symbol = 'X'
        print()
